02/second.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func computer(n []int) int {
+	for i := 0; i < len(n)-1; i += 4 {
+		switch n[i] {
+		case 99:
+			break
+		case 1:
+			n[n[i+3]] = n[n[i+1]] + n[n[i+2]]
```

Comment:
Won't this panic if someone gives it input which is like 7 characters long?

Reply:
What I mean is if the input length mod 4 is not 0.

Reply:
Actually this will break in other cases aswell, since it gets the index from the input.

Reply:
So 1. you should check if `i+3`, `i+2`, `i+1` etc is in the range of the array and subsequently if `n[i+3]`, `n[i+2]`, n[i+1]` is in range.